haskell-library/.github/workflows/check.yml

```diff
@@ -0,0 +1,49 @@
+name: Nix flake check
+on: push
+
+jobs:
+  validate:
+    runs-on: self-hosted
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: xrefcheck
+        run: nix run github:serokell/xrefcheck
+
+      - name: reuse
+        run: nix build -L .#checks.x86_64-linux.reuse-lint
+        # Run step even if the previous one has failed
+        if: success() || failure()
+
+      - name: hlint
+        run: nix build -L .#checks.x86_64-linux.hlint
+        if: success() || failure()
+
+      - name: stylish-haskell
+        run: nix build -L .#checks.x86_64-linux.stylish-haskell
+        if: success() || failure()
+
+      - name: check-trailing-whitespace
+        run: nix build -L .#checks.x86_64-linux.trailing-whitespace
+        if: success() || failure()
+
+      # - name: shellcheck
+      #   run: nix build -L .#checks.x86_64-linux.shellcheck
+      #   if: success() || failure()
+
+  build:
+    runs-on: self-hosted
```

Comment:
Another thing that might be useful is to use a jobs matrix here and build for each GHC version separately. It should be possible to generate a matrix json from one of the previous jobs and use it for building and running tests, take a look at https://github.blog/changelog/2020-04-15-github-actions-new-workflow-features/ for inspiration

Reply:
Currently `nix build` will exit once one of the derivations build in the deps tree has failed, as a result, we won't be able to easily check that it doesn't for e.g. exactly one GHC version